Studio-Lovelies/GG-JointJustice-Unity
# Move and Talk options require IDs

## Summary
Reference Move and Talk choices via internal ID, instead of label text and background image tag, giving more flexibility.

> [!WARNING]  
> This breaks any script that uses the investigation UI **by design**. Cross examination and dialogue should remain unaffected however.

## Before/after screenshots and/or animated gif
### Before
```
+ [Tutorial #Talk #locked]
    -> Tutorial

&UNLOCK_TALK_CHOICE:Tutorial
```
### After
```
+ [Tutorial #id:thetutorial #Talk #locked]
    -> Tutorial

&UNLOCK_TALK_CHOICE:Tutorial    <-- will *not* unlock the choice (but also not fail)

&UNLOCK_TALK_CHOICE:thetutorial <-- will unlock the choice
```



## Testing instructions
1. Open the Game scene
2. Set the `GameStarter` to `InvestigationUI`
3. Play through the talk and move choices and notice no difference
4. All automated tests grouped under `PlayModeTests->Suites->Scripts->InvestigationState` pass
## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  - #456
